### ISSUE TITLE
[FEATURE] Using Jackson Annotations simultaneously: @JsonNaming @JsonAlias

### ISSUE THREAD

**Issue by the-codinator** [FasterXML/jackson-annotations#187]
Java: 11.0.10
Jackson: 2.12.2

We currently have all fields returned by an API in kebab case (eg: `first-name`), which wants to convert to camel case for consistency with other systems (eg: `firstName`).

This is obviously a breaking change, so the right way to do it would be the API including both versions of the field (kebab + camel), giving sufficient time for folks to migrate from the deprecated key name, and then removing the deprecated fields.

As an improvement, to further decouple the API and consumer applications, it would be great to have this support added to `JsonNaming` or `JsonAlias`

Currently, we have `@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)` at the top of our class to have all fields being deserialized from kebab case. Maybe `JsonNaming` can take another param `asAlias = true` to indicate that the naming pattern is an alias only, i.e. both naming conventions should be permitted for deserialization (default, or aliased) and also allow multiple instances of `@JsonNaming` with at most one having `asAlias = false`.

An alternative would be to allow `@JsonAlias` to be applied at the class level, with a `PropertyNamingStrategy` as its value instead.

Or if its not a good idea fitting this into existing annotations, create a `JsonNamingAlias` which can be used like the following:

```java
@JsonNaming(MainPropertyNamingStrategyUsedForSerializationAndDeserialization.class)
@JsonNamingAlias(PropertyNamingStrategies.KebabCaseStrategy.class)
@JsonNamingAlias(AnotherStrategy.class)
class MyPojo { }
```

If there are fields in the json to be deserialized which point to the same key with different naming formats (eg: `{"first-name": "codi1", "firstName": "codi2"}`, then just follow what happens when multiple `JsonAlias` are applied to a field (I believe the precedence is actual property name (or `JsonProperty` if present) followed first matched alias)

**Comment by the-codinator** [comment]
@cowtowncoder tagging you since I think its a good feature to add and don't want to lose it out due to inactivity...

Apologies if you felt this tag to draw attention was spam (let me know and I'll refrain henceforth).

**Comment by cowtowncoder** [comment]
If anyone has time & interest they are free to propose an implementation. This sounds like a rather complicated thing to implement so I don't really have time to look into it any further myself, but maybe someone else has.

**Comment by the-codinator** [comment]
Got it. I'll take a stab at this pretty soon...

**Comment by cowtowncoder** [comment]
Makes sense! Adding of new annotation is simple enough; next step would be to add API method in `AnnotationIntrospector`.
